restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/SmsMessagesDao.java

```diff
@@ -53,4 +54,6 @@
     // Support for filtering of message list result, Issue 1395
     Integer getTotalSmsMessage(SmsMessageFilter filter);
     List<SmsMessage> getSmsMessages(SmsMessageFilter filter);
+
+    List<SmsMessage> findBySmppMessageIdAndDateCreatedGreaterOrEqualThanOrderedByDateCreatedDesc(String smppMessageId, DateTime startDate);
```

Comment:
i would prefer a Filter based API approach as in https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/IncomingPhoneNumbersDao.java#L48

youll see it supports sorting as well https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/entities/IncomingPhoneNumberFilter.java#L34